packages/fuels-test-helpers/src/fuel_bin_service.rs

```
@@ -0,0 +1,222 @@
+            .map(|(_, arg)| arg.to_string()),
+        );
+
+        Ok(args)
+    }
+
+    pub fn write_temp_chain_config_file(&mut self) -> FuelResult<()> {
+        Ok(serde_json::to_writer(
+            &mut self.config_file,
+            &self.config.chain_conf,
+        )?)
+    }
+}
+
+#[derive(Clone, Default, Debug)]
+pub struct SharedState {
+    pub config: Config,
+}
+
+pub struct ServerParams {
+    extended_config: ExtendedConfig,
+}
+
+pub struct FuelService {
+    pub bound_address: SocketAddr,
+    runner: Pin<Box<JoinHandle<()>>>,
+}
+
+impl FuelService {
+    pub async fn new_node(config: Config) -> FuelResult<Self> {
+        let requested_port = config.addr.port();
+
+        let bound_address = match requested_port {
+            0 => get_socket_address(),
+            _ if is_free(requested_port) => config.addr,
+            _ => return Err(Error::IOError(std::io::ErrorKind::AddrInUse.into())),
+        };
+
+        let config = Config {
+            addr: bound_address,
+            ..config
+        };
+
+        let params = ServerParams {
+            extended_config: ExtendedConfig {
+                config,
+                config_file: NamedTempFile::new()?,
+            },
+        };
+
+        let runner = run_node(params).await?;
+
+        Ok(FuelService {
+            bound_address,
+            runner: Box::pin(runner),
+        })
+    }
+
+    pub async fn stop(&self) -> FuelResult<State> {
+        self.runner.abort();
+        Ok(State::Stopped)
+    }
+}
+
+async fn server_health_check(client: &FuelClient) -> FuelResult<()> {
+    let mut attempts = 5;
+    let mut healthy = client.health().await.unwrap_or(false);
+    let between_attempts = Duration::from_millis(300);
+
+    while attempts > 0 && !healthy {
+        healthy = client.health().await.unwrap_or(false);
+        sleep(between_attempts).await;
+        attempts -= 1;
+    }
+
+    if !healthy {
+        panic!("error: Could not connect to fuel core server.")
```

IMO we should return an error here. The function already returns a result